### PR TITLE
Disable adding logs as breadcrumbs to Sentry

### PR DESCRIPTION
**Pull request by wzieba**
To increase user privacy. We should use Encrypted Logging for this, which already contains these logs (+ more)

-----

## To Test:

Not needed - from now on Sentry breadcrumbs won't show application logs. But we still have access to them from Encrypted Logs.

-----

## Regression Notes

1. Potential unintended areas of impact

    - none

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - none

3. What automated tests I added (or what prevented me from doing so)

    - none

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by thomashorta** [comment]
@wzieba I agree with increasing user privacy, just wanted to make 2 suggestions as follow up to merging this PR. 

### Announcement
It would be great if you could post an internal announcement somewhere saying that this log reporting is going away, since I, and I'm pretty sure other Engineers, usually go through Sentry's breadcrumbs and look at those logs in the timeline to analyze issues and user journeys. 

Many times in non-reproducible crashes I was actually only able to figure out what was going on (and even come up with a fix) thanks to the logs being shown as breadcrumbs.

If you could include in that announcement how we should use Encrypted Logging as a replacement and how to find these logs based on a Sentry crash that would be awesome, as I personally don't know how to do it.

### Breadcrumbs improvement
Beyond sending all logs to Sentry as breadcrumbs I don't think we are using the breadcrumbs functionality properly, and removing these logs means that we will probably lose basically all useful breadcrumbs in Sentry since the default ones sent by Sentry's SDK are REALLY basic (it's like: Activity lifecycle events, and exceptions).

So my suggestion is that we come up with some way of improving our code to send relevant breadcrumbs when needed (e.g.: user interactions, navigation, service scheduling/cancelation, notification, etc.), so we can figure out user journeys in Sentry to help with crash analysis.

Here's an interesting recent post about that being done in another one of our products: pbCcaG-3BW-p2

**Comment by wzieba** [comment]
hi there 👋 ! Thanks for sharing those concerns:

> Announcement

Sure! I'll drop the info in our P2 when this PR will be merged.

> Many times in non-reproducible crashes I was actually only able to figure out what was going on (and even come up with a fix) thanks to the logs being shown as breadcrumbs.

I understand there's a great value that comes with logging - encrypted logs should have the same data as breadcrumbs. Actually, I just compared some breadcrumbs vs. encrypted logs, and encrypted logs contain _more_ data, which is helpful when debugging.

> If you could include in that announcement how we should use Encrypted Logging as a replacement and how to find these logs based on a Sentry crash that would be awesome, as I personally don't know how to do it.

This post should be helpful: paaHJt-1ms-p2

> Beyond sending all logs to Sentry as breadcrumbs I don't think we are using the breadcrumbs functionality properly, and removing these logs means that we will probably lose basically all useful breadcrumbs in Sentry since the default ones sent by Sentry's SDK are REALLY basic (it's like: Activity lifecycle events, and exceptions).

I'd actually advise to not use breadcrumbs for any specific user journeys like user interactions or navigation, but rather simply log them via `AppLog`.

In other words: after merging this PR, we don't lose any data - we just stop sharing it with Sentry. The same data (and more) will be now available only via Encrypted Logs.

**Comment by thomashorta** [comment]
> In other words: after merging this PR, we don't lose any data - we just stop sharing it with Sentry. The same data (and more) will be now available only via Encrypted Logs.

Awesome! I believe I used Encrypted Logs once and completely forgot about it. 🤦🏼 Thanks for the reminder!

**Review by thomashorta — APPROVED** [review]
Thanks for checking and answering my concerns! Code-wise it looks good to me so I will approve it.

**Comment by wzieba** [comment]
Thanks @thomashorta ! Merging then :shipit:
